app/Application.scala

```diff
@@ -1,32 +1,36 @@
-import com.gu.contentapi.sanity._
-import com.gu.contentapi.sanity.support.{CloudWatchReporter, InfrequentScheduledTestsFailureHandler, FrequentScheduledTestFailureHandler}
+package com.gu.contentapi.sanity
+
+import com.gu.contentapi.sanity.support.{ CloudWatchReporter, FrequentScheduledTestFailureHandler, InfrequentScheduledTestsFailureHandler }
 import com.gu.contentapi.sanity.utils.QuartzScheduler
 import org.joda.time.DateTime
+import play.api._, play.api.inject.ApplicationLifecycle
+import play.api.mvc.{ ControllerComponents, BaseController }
+import scala.concurrent.Future
 import scala.concurrent.duration._
-import scala.language.postfixOps
-import play.api._
-
-object Global extends GlobalSettings {
 
-  override def onStart(app: Application) {
-    val cloudWatchReporter = CloudWatchReporter(app.configuration)
+class Application(
+  lifecycle: ApplicationLifecycle,
+  cloudWatchReporter: CloudWatchReporter
+) {
 
-    Logger.info("Application has started")
-    QuartzScheduler.start()
-    QuartzScheduler schedule("Frequent Tests", runFrequentTests(cloudWatchReporter)) every (30 seconds)
-    QuartzScheduler schedule("Infrequent Tests", runInfrequentTests(cloudWatchReporter)) at "0 0 9 ? * MON-FRI *"
-  }
-
-  override def onStop(app: Application) {
+  lifecycle.addStopHook { () =>
     Logger.info("Application shutdown...")
-    QuartzScheduler.stop()
+    Future.successful(QuartzScheduler.stop())
   }
 
+  Logger.info("Application has started")
+  QuartzScheduler.start()
+  QuartzScheduler schedule("Frequent Tests", runFrequentTests(cloudWatchReporter)) every (30.seconds)
+  QuartzScheduler schedule("Infrequent Tests", runInfrequentTests(cloudWatchReporter)) at "0 0 9 ? * MON-FRI *"
+
   def runFrequentTests(cloudWatchReporter: CloudWatchReporter): Unit = {
     println(s"=== Starting frequent tests at ${new DateTime()} ===")
     val context = Context(testFailureHandler = FrequentScheduledTestFailureHandler, cloudWatchReporter = cloudWatchReporter)
     val suites = MetaSuites.prodFrequent(context)
-    suites.foreach(_.execute)
+    suites.foreach { s =>
+      s.execute()
+      s.onComplete
```

Comment:
Could this also be achieved by making `SanityTestBase` extend `BeforeAndAfter`?

Reply:
😮 

scalatest documentation leaves a lot to be desired 😓  thanks!